### ISSUE TITLE
bug: `@core/llamaindex` doesn't allow access to various schemas

### ISSUE THREAD

**Issue by parhammmm** [run-llama/LlamaIndexTS#987]
Attempting to import the following from `@core/llamaindex` and getting `Cannot find declarations`

`Document`
`Metadata`
`BaseNode`
`MetadataMode`
`NodeWithScore`

```ts
// Doesn't work:
import { Document } from "@llamaindex/core";

// Doesn't work
import { Document } from "@llamaindex/core/schema";

// Works for import but gives `Error: Missing "./dist/schema" specifier in "@llamaindex/core" package`
import { Document } from "@llamaindex/core/dist/schema";
```

any tips @himself65 ?

P.S. I think this is a good direction, just need to figure out the above

**Comment by himself65** [comment]
change module resolution to node16 or bundler in your tsconfig

**Comment by parhammmm** [comment]
@himself65 thanks! switching to bundler has resolved it

**Comment by himself65** [comment]
nit: always use `bundler` if you are not running code like `tsc && node ./dist/index.js`

**Comment by parhammmm** [comment]
@himself65 I'm trying to find a work around right now but setting to bundler might not always be possible

more context: building or azure functions app

**Comment by himself65** [comment]
> @himself65 I'm trying to find a work around right now but setting to bundler might not always be possible
> 
> more context: building or azure functions app

I don't have much further idea about this, but this is ts old behavior. But whether you update your tsconfig.json, do some patches to make type check happy, or use a bundler to a single js file before releasing to the Azure function app.

maybe related issue: https://github.com/microsoft/TypeScript/issues/46452#issue-1031817996

**Comment by parhammmm** [comment]
@himself65 apologies totally forgot to update you on this. 

That's fair and very reasonable not want to support anything below node18.

In the end I managed to work around everything bit by bit to get the build working. 

The one thing that was frustrating and I figured out in the end was, prettier had to be re-configured too, renaming it's config from `.prettierrc.js` to `.prettierrc.json` forced it to use modules in the end
